### PR TITLE
Update the Docker configuration

### PR DESCRIPTION
**Pull request by sroze**
Update the Docker configuration to remove the host-specific configuration and use [immutable images](http://sroze.io/2015/12/23/immutable-docker-images/).

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# Install composer
+RUN curl -sS https://getcomposer.org/installer | php \
+    && mv composer.phar /usr/bin/composer
+
+# Add the application
+ADD . /app
+WORKDIR /app
+
+# Remove cache and logs if some and fixes permissions
+RUN ((rm -rf app/cache/* && rm -rf app/logs/*) || true) \
```
Comment (jderusse) [inline Dockerfile]: What'd about using .dockerignore

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    && chown www-data . app/cache app/logs
+
+# Install dependencies
+RUN composer install -o
```
Comment (jderusse) [inline Dockerfile]: Should be run as www-data to avoid permissions issues with files generated by scripts.

File: Dockerfile (hunk 1 above)
Comment (sroze) [inline Dockerfile]: `.dockerignore` is not recursive :(

**Comment by jderusse** [comment]
Can run a cache warmup too

**Comment by dunglas** [comment]
:+1:

File: Dockerfile (hunk 1 above)
Comment (jderusse) [inline Dockerfile]: Even with `**` pattern ?
I never used it, but I'm really surprised we can no ignore complete directory. :(

Comment (sroze) [inline Dockerfile]: This was [merged](https://github.com/docker/docker/pull/17090) for Docker 1.10 only, while most of the users still don't use it.

File: Dockerfile (hunk 2 above)
Comment (sroze) [inline Dockerfile]: When doing that, you basically have to update your `/etc/passwd` to be able to `su - www-data`, Then, with the `.:/app` volume in the `docker-compose.yml` file, your existing files in `vendor/` do not belong to `www-data` so you need to chown everything.

Don't get me wrong I'm not against. Still, do you think that's worth doing it? I like the idea of warming up the cache though.

Comment (jderusse) [inline Dockerfile]: May be, just move the chown www-data after the composer install?

**Comment by sroze** [comment]
@jderusse I've added the cache warmup, good idea!

**Pull request merged**